src/rgw/rgw_lc.cc

```diff
@@ -1073,22 +1073,17 @@ class LCOpAction_CurrentExpiration : public LCOpAction {
       return false;
     }
     if (o.is_delete_marker()) {
-      if (oc.next_key_name) {
-	std::string nkn = *oc.next_key_name;
-	if (oc.next_has_same_name(o.key.name)) {
-	  ldpp_dout(dpp, 7) << __func__ << "(): dm-check SAME: key=" << o.key
-			   << " next_key_name: %%" << nkn << "%% "
-			   << oc.wq->thr_name() << dendl;
-	  return false;
-	} else {
-	  ldpp_dout(dpp, 7) << __func__ << "(): dm-check DELE: key=" << o.key
-			   << " next_key_name: %%" << nkn << "%% "
-			   << oc.wq->thr_name() << dendl;
-        *exp_time = real_clock::now();
-        return true;
-	}
+      if (oc.next_has_same_name(o.key.name)) {
+        ldpp_dout(dpp, 7) << __func__ << "(): dm-check SAME: key=" << o.key
+                          << " next_key_name: %%" << *oc.next_key_name << "%% "
+                          << oc.wq->thr_name() << dendl;
+        return false;
       }
-      return false;
```

Comment:
This is the important bit, right?  So that it falls through to the next code?

Reply:
yes it falls through to the next code intentionally. 
if it's delete-marker, we return false, that is to skip the lc, if there is another entry with the same name, otherwise return true.